drivers/include/sdp3x.h

```diff
@@ -0,0 +1,204 @@
+ * @}
+ */
+
+#ifndef SDP3X_H
+#define SDP3X_H
+
+#include "saul.h"
+#include "xtimer.h"
+#include "periph/i2c.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   SDP3x models
+ */
+typedef enum {
+    SDP3x_31 = 1,
+    SDP3x_32 = 2
+} sdp3x_model_t;
+
+/**
+ * @brief   Device initialization parameters
+ */
+typedef struct {
+    i2c_t i2c_dev;                  /**< I2C device which is used */
+    uint8_t i2c_addr;               /**< I2C address */
+    sdp3x_model_t model;            /**< Model Number */
+    gpio_t irq_pin;                 /**< IRQ pin to check data ready */
+} sdp3x_params_t;
+
+/**
+ * @brief   Device descriptor for the SDP3x sensor
+ */
+typedef struct {
+    sdp3x_params_t params;    /**< Device initialization parameters */
+} sdp3x_t;
+
+/**
+ * @brief   Measurement from SDP3x sensor
+ */
+typedef struct {
+    double differential_pressure;  /**< Differential Pressure measured*/
+    double temperature;            /**< Temperature measured*/
+    int16_t dp_scale_factor;       /**< Scale factor for measurement*/
+} sdp3x_measurement_t;
+
+/**
+ * @brief   Status and error return codes
+ */
+enum {
+    SDP3x_OK = 0,             /**< all went as expected */
+    SDP3x_CRCERROR = -1,      /**< CRC-Check failed */
+    SDP3x_NODATA = -2,        /**< No Data available */
+    SDP3x_IOERROR = -3,       /**< I/O error */
+    SDP3x_WRONGSIZE = -4      /**< Wrong size of array for method */
+};
+
+/** Initialize SPD3x
+ *
+ *      @param dev         sdp3x device
+ *      @param params      sdp3x device params
+ *
+ *      @return            1 if device started
+ */
+int SDP3x_init(sdp3x_t *dev, const sdp3x_params_t *params);
+
+/** Start Continuous Measuring
+ *
+ *      @param dev          sdp3x device
+ *      @param tempComp     Temperature compensation:
+ *                          1 = Mass flow, 2 = Differential pressure
+ *      @param avg          Choose method for averaging values:
+ *                          0 = No averaging, 1 = Averaging till read
+ *      @param interval     Interval in microseconds in which values are read
+ *      @param result       Values are stored in this struct
+ *
+ *      @return             1 if device started
+ */
+int8_t SDP3x_start_continuous(const sdp3x_t *dev, uint8_t tempComp, uint8_t avg,
+       uint32_t interval, sdp3x_measurement_t *result);
+
+/** read temperature and differential pressure
+ *
+ *      @param result       struct to store result
+ *      @param dev          sdp3x device
+ *
+ *      @return             1 on success
+ */
+
+int8_t SDP3x_read_continuous(sdp3x_measurement_t *result, sdp3x_t *dev);
+
+/** Read current temperature value
+ *
+ *      @param dev           sdp3x device
+ *      @param tempComp      Temperature compensation:
+ *                           1 = Mass flow, 2 = Differential pressure
+ *      @param clkStretching Activates clock stretching for measurement:
+ *                           0 = Deactivated, 1 = Activated
```

Comment:
How would a user of the API choose when to activate / deactivate that?
Does the user need that choice at all?